internal/provisioner/plain/controllers/bundle_controller.go

```diff
@@ -359,7 +359,7 @@ func getObjects(bundleFS fs.FS) ([]client.Object, error) {
 	}
 	for _, e := range entries {
 		if e.IsDir() {
-			continue
+			return nil, fmt.Errorf("subdirectories are not allowed in the bundle image filesystem: found %q", e.Name())
```

Comment:
One more nit on this error message: we're specifically disallowing subdirectories _within the manifests directory_.

```suggestion
			return nil, fmt.Errorf("subdirectories are not allowed within the %q directory of the bundle image filesystem: found %q", manifestsDir, filepath.Join(manifestsDir, e.Name()))
```

Reply:
OK.  It's clearer.  I'll update. Thanks!